python/audio_dsp/models/signal_chain.py

```diff
@@ -193,6 +193,16 @@ class SwitchStereo(StageModel):
     op_type: Literal["SwitchStereo"] = "SwitchStereo"
     parameters: SwitchParameters = Field(default_factory=SwitchParameters)
 
+    @model_validator(mode="after")
+    def set_max_outputs(self):
+        """Set the maximum number os switch positions."""
+        max_val = len(self.placement.input)
+        type(self.parameters).model_fields["position"].metadata.append(
```

Comment:
Injecting metadata after the model already validated had no effect.
We should have something like this instead
```suggestion
        if (self.parameters.position > max_val):
            raise ValueError(
                f'Switch position should be less than or equal to {max_val}'
            )
```